test/library/standard/Random/ferguson/pcg-rand-check.chpl

```diff
@@ -15,54 +15,54 @@ writeln("Checking 32-bit RNG seq 1");
 {
   var rs = createRandomStream(seed = seed, parSafe=false,
                             eltType = uint(32), algorithm=RNG.PCG);
-  for i in 1..6 {
+  for e32 in expect32_1 {
     //writef("%xu\n", rs.RandomStreamPrivate_rng_states(1));
     var got = rs.getNext();
     if verbose then writef("%xu\n", got);
-    assert( got == expect32_1[i] );
+    assert( got == e32 );
   }
 
   rs.skipToNth(1);
 
-  for i in 1..6 {
+  for e32 in expect32_1 {
     //writef("%xu\n", rs.RandomStreamPrivate_rng_states(1));
     var got = rs.getNext();
 
     if verbose then writef("%xu\n", got);
-    assert( got == expect32_1[i] );
+    assert( got == e32 );
   }
 
-  for i in 1..6 {
+  for (i, e32) in zip(1..6, expect32_1) {
     var got = rs.getNth(i);
 
     if verbose then writef("%xu\n", got);
-    assert( got == expect32_1[i] );
+    assert( got == e32 );
   }
 }
 
 {
   var rs = createRandomStream(seed = seed, parSafe=false,
                             eltType = int(32), algorithm=RNG.PCG);
-  for i in 1..6 {
+  for e32 in expect32_1 {
     //writef("%xu\n", rs.RandomStreamPrivate_rng_states(1));
     var got = rs.getNext();
     if verbose then writef("%xu\n", got);
-    assert( got:uint(32) == expect32_1[i] );
+    assert( got:uint(32) == e32 );
   }
 
   rs.skipToNth(1);
 
-  for i in 1..6 {
+  for e32 in expect32_1 {
     //writef("%xu\n", rs.RandomStreamPrivate_rng_states(1));
     var got = rs.getNext();
     if verbose then writef("%xu\n", got);
-    assert( got:uint(32) == expect32_1[i] );
+    assert( got:uint(32) == e32 );
   }
 
-  for i in 1..6 {
+  for (i, e32) in zip(1..6, expect32_1) {
     var got = rs.getNth(i);
```

Comment:
huh... does this vaguely imply that random streams should also be 0-based?

Reply:
I think they probably should be for consistency, and can never remember whether one of my  PRs did that or not.

Reply:
I filed https://github.com/chapel-lang/chapel/issues/15418 to not forget this.

Reply:
I thumbs-upped the issue :)